src/punx/github_handler.py

```diff
@@ -107,32 +107,36 @@ def __init__(self):
         self.sha = None
         self.zip_url = None
         self.last_modified = None
-    
+
     def connect_repo(self, repo_name=None, token=None):
         """
         connect with the GitHub repository
-        
+
         :param str repo_name: name of repository in https://github.com/nexusformat (default: *definitions*)
         :returns bool: True if using GitHub credentials
         """
         repo_name = repo_name or self.appName
-        
+
+        token = get_BasicAuth_credentials() if token is None else token
```

Comment:
with above refactor so `get_GitHub_credentials()` returns a dict:

```python
token = get_GitHub_credentials()["token"] if token is None else token
```

Reply:
Why should get_GitHub_credentials return a dictionary? Only the token is ever used.

Reply:
That's true within *this* software.  But on my system, the exact same file is used in other contexts that need the additional content and formatting.  A change here would compel revision to the other software.

Why should the current interface change?